classes/affiliation/Repository.php

```diff
@@ -0,0 +1,302 @@
+<?php
+
+/**
+ * @file classes/affiliation/Repository.php
+ *
+ * Copyright (c) 2024 Simon Fraser University
+ * Copyright (c) 2024 John Willinsky
+ * Distributed under the GNU GPL v3. For full terms see the file docs/COPYING.
+ *
+ * @class Repository
+ *
+ * @brief A repository to find and manage affiliations.
+ */
+
+namespace PKP\affiliation;
+
+use APP\author\Author;
+use APP\core\Request;
+use APP\facades\Repo;
+use APP\submission\Submission;
+use Illuminate\Support\Facades\App;
+use PKP\context\Context;
+use PKP\plugins\Hook;
+use PKP\services\PKPSchemaService;
+use PKP\validation\ValidatorFactory;
+
+class Repository
+{
+    public DAO $dao;
+
+    /** The name of the class to map this entity to its schema */
+    public string $schemaMap = maps\Schema::class;
+
+    protected Request $request;
+
+    /** @var PKPSchemaService<Affiliation> */
+    protected PKPSchemaService $schemaService;
+
+    public function __construct(DAO $dao, Request $request, PKPSchemaService $schemaService)
+    {
+        $this->dao = $dao;
+        $this->request = $request;
+        $this->schemaService = $schemaService;
+    }
+
+    /** @copydoc DAO::newDataObject() */
+    public function newDataObject(array $params = []): Affiliation
+    {
+        $object = $this->dao->newDataObject();
+        if (!empty($params)) {
+            $object->setAllData($params);
+        }
+        return $object;
+    }
+
+    /** @copydoc DAO::exists() */
+    public function exists(int $id, ?int $authorId = null): bool
+    {
+        return $this->dao->exists($id, $authorId);
+    }
+
+    /** @copydoc DAO::get() */
+    public function get(int $id, ?int $authorId = null): ?Affiliation
+    {
+        return $this->dao->get($id, $authorId);
+    }
+
+    /** @copydoc DAO::getCollector() */
+    public function getCollector(): Collector
+    {
+        return App::make(Collector::class);
+    }
+
+    /**
+     * Get an instance of the map class for mapping affiliations to their schema.
+     */
+    public function getSchemaMap(): maps\Schema
+    {
+        return app('maps')->withExtensions($this->schemaMap);
+    }
+
+    /**
+     * Validate properties for an affiliation
+     *
+     * Perform validation checks on data used to add or edit an affiliation.
+     *
+     * @param Author|null $author Author being edited. Pass `null` if creating a new author
+     * @param array $props A key/value array with the new data to validate
+     * @param Submission $submission The context's supported locales
+     * @param Context $context The context's primary locale
+     *
+     * @return array A key/value array with validation errors. Empty if no errors
+     *
+     * @hook Affiliation::validate [[&$errors, $object, $props, $submission, $context]]
+     */
+    public function validate(?Author $author, array $props, Submission $submission, Context $context): array
```

Comment:
Maybe to have the syntax as in all validate functions: to validate only one affiliation and to have the first parameter Affiliation object that could be null ?